impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultPluginXmlFactory.java

```diff
@@ -45,37 +48,47 @@
 @Named
 @Singleton
 public class DefaultPluginXmlFactory implements PluginXmlFactory {
+
     @Override
     public PluginDescriptor read(@Nonnull XmlReaderRequest request) throws XmlReaderException {
         nonNull(request, "request");
         Path path = request.getPath();
         URL url = request.getURL();
         Reader reader = request.getReader();
         InputStream inputStream = request.getInputStream();
-        if (path == null && url == null && reader == null && inputStream == null) {
+        if (inputStream == null && reader == null && path == null && url == null) {
             throw new IllegalArgumentException("path, url, reader or inputStream must be non null");
         }
+        return read(request, inputStream, reader, path, url);
+    }
+
+    private static PluginDescriptor read(
+            XmlReaderRequest request, InputStream inputStream, Reader reader, Path path, URL url) {
         try {
             PluginDescriptorStaxReader xml = new PluginDescriptorStaxReader();
             xml.setAddDefaultEntities(request.isAddDefaultEntities());
             if (inputStream != null) {
-                return xml.read(inputStream, request.isStrict());
+                return read(request, xml, inputStream);
             } else if (reader != null) {
                 return xml.read(reader, request.isStrict());
             } else if (path != null) {
                 try (InputStream is = Files.newInputStream(path)) {
-                    return xml.read(is, request.isStrict());
-                }
-            } else {
-                try (InputStream is = url.openStream()) {
-                    return xml.read(is, request.isStrict());
+                    return read(request, xml, is);
                 }
             }
-        } catch (Exception e) {
```

Comment:
IllegalCatch

- https://github.com/apache/maven-shared-resources/pull/73
- https://checkstyle.sourceforge.io/checks/coding/illegalcatch.html

> we shouldn't be catching raw exceptions here; probably fix that first

https://github.com/apache/maven/pull/2306#discussion_r2081558306

Reply:
would be simple removal of broad `Exception` and downgrade `catch` to whats actually thrown (going on) `IOException | XMLStreamException` @timtebeek